my-code/significa/src/pages/Home.js

```diff
@@ -0,0 +1,190 @@
+import React, { useState, useEffect } from "react";
+import { Link } from "@reach/router";
+
+import Layout from "../components/layout";
+import MovieCard from "../components/MovieCard";
+import magnifier from "../assets/icons/icon-magnifier-grey.svg";
+import illustrationSm from "../assets/illustrations/illustration-empty-state.png";
+import illustrationLg from "../assets/illustrations/illustration-empty-state@2x.png";
+import "./Home.css";
+
+const API_KEY = `${process.env.REACT_APP_API_KEY}`;
+
+const Home = () => {
+  /* set variables that use state */
+  const [query, setQuery] = useState("");
+  const [empty, setEmpty] = useState(true);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [movies, setMovies] = useState([]);
+
+  /* function that handles change in the search bar */
+  const handleChange = (e) => {
+    /* console.log(e.target.value); */
+    if (e.target.value.trim() === "") {
+      setEmpty(true);
+    } else {
+      setEmpty(false);
+    }
+    return setQuery(e.target.value.trim());
+  };
+
+  /* fetch list of movies from the api when query changes */
+  useEffect(() => {
+    let mounted = true;
+
+    /* function that gets list of movies from api */
+    const getList = async (requestUrl) => {
+      try {
+        let response = await fetch(requestUrl);
+        let json = await response.json();
+        if (mounted) {
+          setIsLoading(false);
+          setMovies(json.Search);
+        }
+      } catch (error) {
+        setIsLoading(false);
+        setError(error);
+      }
+    };
+
+    const requestUrl = encodeURI(
+      `https://www.omdbapi.com/?s=${query}&apikey=${API_KEY}`
+    );
+    setIsLoading(true);
+    getList(requestUrl);
+
+    return () => (mounted = false);
+  }, [query]);
```

Comment:
Make sure to only request to the server when the user is ready. I mean, it's requesting on every keydown event, so there is a lot of unnecessary requests here. For instance, I've searched for 007 and the network has 3 requests, but only the last one is helpful. Did you get my point?

To solve it, there are some common approaches to do it, like `debounce` and `throttle`. Try to think which one fits better for this problem.